cpp/include/raft/sparse/convert/csr.cuh

```diff
@@ -102,6 +105,26 @@ void adj_to_csr(raft::resources const& handle,
   detail::adj_to_csr(handle, adj, row_ind, num_rows, num_cols, tmp, out_col_ind);
 }
 
+/**
+ * @brief Converts a bitmap matrix into unsorted CSR format matrix.
+ *
+ * @tparam       bitmap_t    Underlying type of the bitmap.
+ * @tparam       index_t     Indexing type used.
+ * @tparam       value_t     Data type of CSR
+ * @tparam       nnz_t       Type of CSR
+ *
+ * @param[in]    handle      RAFT handle
+ * @param[in]    bitmap      input raft::bitmap_view
+ * @param[out]   csr         output raft::device_csr_matrix_view
+ */
+template <typename bitmap_t, typename index_t, typename value_t, typename nnz_t>
+void bitmap_to_csr(raft::resources const& handle,
+                   raft::core::bitmap_view<bitmap_t, index_t> bitmap,
+                   raft::device_csr_matrix_view<value_t, index_t, index_t, nnz_t> csr)
```

Comment:
Our sparse APIs discern between `strucure owning` and `structure preserving`. The former means that the nnz is not yet known when the user creates the object. The latter means that nnz is known upon creation and any processing functions do not have the freedom to change it. One of the reasons we discern between these two is so that upon invoking a function, the user knows whether the function is supposed to compute the sparsity in addition to populating the resulting non-zero weights/elements. It doesn't look like we make that distinction here, yet we are calculating and computing the sparsity in the implementation.

Reply:
Does it mean we should calculate/get the `nnz` of bitmap and reallocate/allocate values by it for `csr` in the API?